packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -357,6 +345,12 @@ abstract contract CrossDomainMessenger is
             "CrossDomainMessenger: message has already been relayed"
         );
 
+        // Another message is being relayed, so we just store this for future replay.
+        if (xDomainMsgSender != Constants.DEFAULT_L2_SENDER) {
+            failedMessages[versionedHash] = true;
```

Comment:
This needs to also emit an event so we can hold the invariant that an event is emitted each time `failedMessages` has a value set in it